### PR TITLE
config: Add enable_virtual_host_style, default false

### PR DESCRIPTION
**Pull request by aprilyang**
@Xuanwo @markduan

### CODE REVIEW

File: src/__snapshots__/build.test.js.snap [hunk 1]
```diff
@@ -47,3 +47,11 @@ Object {
   "uri": "https://pek3a.qingstor.com/test_bucket/test_object.jpg?acl&upload_id=test_upload_id",
 }
 `;
+
+exports[`parseVirtualHostRequestURI 1`] = `
```
Comment (Xuanwo) [inline src/__snapshots__/build.test.js.snap:51]: These files should be removed from git repo.

File: src/__snapshots__/sign.test.js.snap [hunk 2]
```diff
@@ -4,6 +4,8 @@ exports[`getAuthorization 1`] = `"L6q5iKoyWbQky0T5/MYELz5jAFecGbvLDyjLdz6fClw="`
 
 exports[`getCanonicalizedResource 1`] = `"/test_bucket/test_object.jpg?acl&upload_id=test_upload_id"`;
 
+exports[`getCanonicalizedResource by virtual host 1`] = `"/test_bucket/test_object.jpg?acl&upload_id=test_upload_id"`;
```
Comment (Xuanwo) [inline src/__snapshots__/sign.test.js.snap:7]: ditto

File: src/__snapshots__/build.test.js.snap (hunk 1 above)
Comment (aprilyang) [reply src/__snapshots__/build.test.js.snap:51]: https://jestjs.io/docs/en/snapshot-testing

File: src/__snapshots__/sign.test.js.snap (hunk 2 above)
Comment (aprilyang) [reply src/__snapshots__/sign.test.js.snap:7]: https://jestjs.io/docs/en/snapshot-testing

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**